<launch>

	<!-- Agregar camera_info -->
	<!-- recibe camera/depth/image y entrega camera/depth/camera_info -->
	<node pkg="opt_utils" type="add_camera_info" name="add_camera_info">
	<remap from="camera/depth/image" to="bender/sensors/rgbd_head/depth/image_raw"/>
	<remap from="camera/depth/camera_info" to="camera/depth/camera_info" />
	</node>

	<!-- Imagen de disparidad -->
	<node pkg="nodelet" type="nodelet" name="depth_to_disparity" args="standalone depth_image_proc/disparity">
	<remap from="left/image_rect" to="bender/sensors/rgbd_head/depth/image" />
	<remap from="right/camera_info" to="camera/depth/camera_info" />
	<remap from="left/disparity" to="camera/depth/disparity" />
	</node> 

	<!-- Launch ground based people detection node -->
	<node pkg="detection" type="ground_based_people_detector" name="ground_based_people_detector" output="screen" required="true">
    <rosparam command="load" file="$(find detection)/conf/ground_based_people_detector_kinect1.yaml" /> 
    <param name="classifier_file" value="$(find detection)/data/HogSvmPCL.yaml"/>	
		<param name="ground_estimation_mode" value="3" />
    <param name="pointcloud_topic" value="/bender/sensors/rgbd_head/depth/points"/>
		<param name="camera_info_topic" value="/bender/sensors/rgbd_head/rgb/camera_info" />
    <param name="rate" value="60.0"/>  
		<param name="lock_ground" value="true" />
		<param name="background_subtraction" value="false" />
  </node>

	<!-- Launch HaarDispAda Node -->
	<node name="HaarDispAdaNode" pkg="detection" type="HaarDispAda174" output="screen">	
     <rosparam command="load" file="$(find detection)/conf/haar_disp_ada_detector.yaml" /> 
     <remap from="Color_Image" to="/bender/sensors/rgbd_head/rgb/image_raw"/>	
     <remap from="Disparity_Image" to="/camera/depth/disparity"/>
     <remap from="input_detections" to="/ground_based_people_detector/detections"/>
     <param name="classifier_file" value="$(find detection)/data/HaarDispAda.xml"/>
		<param name="use_disparity" value="true" />
   </node> 

	<!-- Launch the tracking node -->
	<arg name="input_topic" value="/detector/detections" />      
  <node pkg="tracking" type="tracker" name="tracker_node" output="screen">
    <remap from="~input" to="$(arg input_topic)" />
    <rosparam command="load" file="$(find tracking)/conf/tracker.yaml" /> 
    <rosparam command="load" file="$(find detection)/conf/haar_disp_ada_detector.yaml" /> 
  </node> 

	<!-- Re-publishing at fixed rate / filtering node -->
  <node pkg="tracking" type="moving_average_filter" name="moving_average_filter_node" output="screen">
    <remap from="~input" to="/tracker/tracks" />
    <remap from="~output" to="/tracker/tracks_smoothed" />
    <remap from="~markers_array" to="/tracker/markers_array_smoothed" />
    <remap from="~history" to="/tracker/history_smoothed" />
    <rosparam command="load" file="$(find tracking)/conf/moving_average_filter.yaml" />
  </node> 

	<!-- Launch follower node -->
	<node pkg="opt_follow" type="depth_follower" name="follower" output="screen" required="true">
    <param name="get_feature_only_start" value="false"/>	
		<param name="nro_only_start" value="10" />
		<param name="target_frames" value="5" />
		<param name="target_dist_thr" value="0.15" />
		<param name="training_factor" value="0.85" /> <!-- 0.55 -->
		<remap from="depth_image" to="/bender/sensors/rgbd_head/depth_registered/image_raw" />
	</node>

</launch>
